services/src/main/java/org/keycloak/broker/oidc/OIDCIdentityProvider.java

```diff
@@ -715,4 +721,32 @@ protected BrokeredIdentityContext exchangeExternalImpl(EventBuilder event, Multi
             throw new ErrorResponseException(OAuthErrorException.INVALID_TOKEN, "invalid token type", Response.Status.BAD_REQUEST);
         }
     }
+
+    @Override
+    protected UriBuilder createAuthorizationUrl(AuthenticationRequest request) {
+        UriBuilder uriBuilder = super.createAuthorizationUrl(request);
+        String nonce = UUID.randomUUID().toString();
```

Comment:
Should we use https://github.com/keycloak/keycloak/blob/master/server-spi-private/src/main/java/org/keycloak/models/utils/KeycloakModelUtils.java#L81 instead of UUID? SecureRandom seems more appropriate for a nonce than an UUID

Reply:
UUID uses SecureRandom behind the scenes. But I can change to reuse the same code from model utils.

Reply:
And that was the code we have previously, so I did not change this specific part.

Reply:
UUID is more expensive to generate than a SecureRandom, and since it doesn't have to be unique, just secure, there's no need to use UUID. Why not just use KeycloakModelUtils directly? It's what we use for example to generate client secrets.

Reply:
@stianst done

Reply:
Created https://issues.redhat.com/browse/KEYCLOAK-12891.